### ISSUE TITLE
Test performance and hyperparameter evaluation

### ISSUE THREAD

**Issue by twoertwein** [JaydenZeng/EMMR#2]
Thank you for sharing the code from your paper!

You report the hyperparameter for your approach in the appendix and also in [this repository](https://github.com/JaydenZeng/EMMR/blob/13e9b13e3a5827d66f45f122b1e37d4ba93c9052/Settings.py#L1). I was wondering whether this repository also contains the code which determined those hyper-parameters. Typically, you train many models and determine the best hyperparameter based on the best validation performance.

Slightly related to the hyper-parameters: in `train.py` you train the model with the reported hyper-parameter over many epochs and then evaluate the test performance. Based on the [code](https://github.com/JaydenZeng/EMMR/blob/13e9b13e3a5827d66f45f122b1e37d4ba93c9052/train.py#L130) it seems that you iterate over all model checkpoints (every n-th epoch), assess their performance on the test set, and then report the best test performance. Is that the code used to determine the test performance reported in the paper?
